.github/pull_request_template.md

```diff
@@ -1,10 +1,22 @@
-**Related Issue(s)**:  ...
+### Proposed Changes
+ <!--- In case of a bug: Describe what caused the issue and how you solved it-->
 
-**Proposed changes**:
-- ...
+### Related Issues
+- fixes #issue-number
+
+### How Did You Test It?
+<!-- unit tests, integration tests, manual verification, instructions for manual tests -->
+
+### Notes for the Reviewer
+<!-- E.g. point out section where the reviewer  -->
+
+### Screenshots (optional)
+<!-- May be added to illustrate the changes -->
 
 ## Pre-flight checklist
-- [ ]  I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md)
+- [ ] I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md)
 - [ ] I have [enabled actions on my fork](https://github.com/deepset-ai/haystack/blob/master/CONTRIBUTING.md#forks)
 - [ ] If this is a code change, I added tests or updated existing ones 
 - [ ] If this is a code change, I updated the docstrings
+- [ ] I've used the [conventional commit convention](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title
+- [ ] I have updated the referenced issue with new insights and changes
```

Comment:
Follow the format of the previous lines, this line could maybe start with something like "If this PR does not close the referenced issue," or other cases. Thereby we make clear that there might be PRs that do not need to update the referenced issue

Reply:
@julian-risch Would it be okay to repost your comment on the Notion page for the handbook? We've continued iterating on the template [there](https://www.notion.so/deepsetai/Version-control-PR-review-and-approval-6775940fe3e644dd936c6f489bc7d3a8?d=759038f618a948e8a148f3d65e295e1e#ce789182832a4f8a8d598814876fe74a)